platform/api.templates/src/org/netbeans/api/templates/CreateFromTemplateHandler.java

```diff
@@ -59,4 +64,171 @@
+                            state = 0;
+                            nested++;
+                        }
+                        break;
+                    case '}': // NOI18N
+                        if (state == 0) {
+                            if (nested-- == 0) {
+                                break A;
+                            }
+                        }
+                        break;
+                }
+                endPos++;
+            }
+            if (endPos >= expression.length()) {
+                pos += 2;
+                continue;
+            }
+            String token = expression.substring(pos + 2, endPos);
+            String defValue = null;
+            
+            int colon = token.indexOf(':'); // NOI18N
+            if (colon != -1) {
+                defValue = token.substring(colon + 1);
+                token = token.substring(0, colon);
+            }
+            
+            if (sb == null) {
+                sb = new StringBuilder();
+            }
+            Object v = parameters.get(token);
+            if (v == null) {
+                v = defValue;
+            }
+            if (v == null) {
+                pos += 2;
+                continue;
+            }
+            sb.append(expression.substring(start, pos)).append(v.toString());
+            start = endPos + 1;
+            pos = start;
+        }
+        if (sb == null) {
+            return expression;
+        }
+        if (start < expression.length()) {
+            sb.append(expression.substring(start));
+        }
+        
+        return mapParameters(sb.toString(), parameters);
+    }
+    
+    private static final String PROP_TEMPLATE = "template"; // NOI18N
+
+    /**
+     * Prefix for template-controlling attributes. Such attributes will not be copied
+     * during template instantiation.
+     */
+    private static final String ATTR_TEMPLATE_PREFIX = "template.";
+    
+    /**
+     * Copies template attributes over to the created file. Copies over from the source (template) to the target (usually
+     * the produced file), except attributes that parametrize template creation. By default, attributes that are prefixed
+     * with {@code "template."} are not copied. As the operation is not atomic, the method should be called in a 
+     * {@link org.openide.filesystems.FileSystem#runAtomicAction(org.openide.filesystems.FileSystem.AtomicAction)}.
+     * The {@code "template"} attribute is never copied.
+     * 
+     * @param h the Handler responsible for file creation, usually the calling handler.
+     * @param from the original file
+     * @param to the target file
+     * @throws IOException if an I/O error occurs.
+     */
+    public static void copyAttributesFromTemplate(CreateFromTemplateHandler h, FileObject from, FileObject to) throws IOException {
+        // copy attributes; some attributes are filtered by FileSystems API already
+        FileUtil.copyAttributes(from, to);
```

Comment:
Consider expanding the `copyAttributes` method with a `Predicate<String> attributeName` rather than copying everything and then cleaning it up.